### PR TITLE
Compensate for breaking change to TypeImpl

### PR DESCRIPTION
**Pull request by nielsenko**
PseudoType inherits from TypeImpl that introduced an abstract
element2 getter and deprecated element. This PR compensate for that.

It also updated all deps to latest major versions.

### CODE REVIEW

File: build.yaml [hunk 1]
```diff
@@ -11,6 +11,9 @@ targets:
       realm_generator|realm_object_builder:
         generate_for:
           include:
```
Comment (desistefanova) [inline build.yaml:13]: Why do we need to include test and example folder?

**Review by desistefanova — APPROVED** [review]

File: build.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ targets:
...
+            - test/**.dart
+            - example/**.dart
```
Comment (desistefanova) [inline build.yaml:15]: Why do we need to include test and example folder?

Comment (nielsenko) [reply build.yaml:15]: This is more to ensure that only test and example folders are processed.

Comment (desistefanova) [reply build.yaml:15]: What about "flutter\realm_flutter\example"? It should be also included, I think.

Comment (nielsenko) [reply build.yaml:15]: That is a project of its own with its own pubspec.yaml and buiild.yaml. You should run the generator in the directory.

Comment (desistefanova) [reply build.yaml:15]: Yes, right.

**Pull request merged**